### ISSUE TITLE
Scheduler is not asynchronous ?

### ISSUE THREAD

**Issue by danielsawan** [ninjaframework/ninja#375]
I have ClassA containing

```
@Schedule(delay = 120, initialDelay = 0, timeUnit = TimeUnit.SECONDS)
public void task1() throws Exception {
```

And ClassB containing 

```
@Schedule(delay = 120, initialDelay = 0, timeUnit = TimeUnit.SECONDS)
public void task2() throws Exception {
```

And on the Module.java Class

```
bind(ClassA.class);
bind(ClassB.class);
```

It seems like task2 never start if task1 is not finished. I also tried to put both tasks on the same Java tasks files and still the same issue.

Am i doing something wrong ? Is it possible to have "Threaded" Tasks ?

**Comment by raphaelbauer** [comment]
It has nothing to do with async. But your observation is right. There is a single thread responsible for scheduling all annotated methods in Ninja. So if you do blocking things you have to offload the work to another thread, so that the scheduler can continue scheduling things :)

The code is here: Scheduler.java - not too complicated imho.

Let us know if you think that there is documentation missing. Or if you got any ideas how to improve the scheduler...

**Comment by danielsawan** [comment]
A simple way to improve the scheduler is to replace in Scheduler.java

```
executor = Executors.newSingleThreadScheduledExecutor();
```

by

```
executor = Executors.newScheduledThreadPool(100);
```

Or Maybe there is a reason i miss to use the SingleThreadScheduledExecutor...

There isn't really a solution in http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html

Don't know why we must specify a fixed poolsize but maybe a highest number  (higher than 100) would even be better. Even if it is not a perfect i think it is still better then only one as it is actualy.

A better solution i think is to implement a quartz module.

It is not really clear in the documentation btw. I think it should be mentionned somewhere that the execution is not parallelized.

**Comment by raphaelbauer** [comment]
This may be one solution to "fix" or relax the problem. Another one is to simply add documentation that the user should make sure to run long running task in his own threadExectuorService...

There should also be a configuration variable in application.conf to configure the # of threads for the scheduler...

**Comment by danielsawan** [comment]
Yes why not a configuration variable. At the moment we don't have the choice it is only 1. Making it configurable should be better.

**Comment by danielsawan** [comment]
I created a pull request if you want to have a look : https://github.com/ninjaframework/ninja/pull/376

**Comment by jfendler** [comment]
@danielsawan you might want to have a look at a new [Ninja-Quartz](https://github.com/FendlerConsulting/ninja-quartz) module.

It's in a very early stage, but might already give you sufficient results for parallel jobs. If you try it out, I'd be happy for every piece of feedback.

**Comment by danielsawan** [comment]
hi @jfendler i saw your pull request and i already put it on my tocheck list :).
I appreciate your notification !
Thanks :)
